homeassistant/components/reolink/camera.py

```diff
@@ -0,0 +1,63 @@
+"""This component provides support for Reolink IP cameras."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.camera import Camera, CameraEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+from .host import ReolinkHost
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink IP Camera."""
+    host: ReolinkHost = hass.data[DOMAIN][config_entry.entry_id].host
+
+    cameras = []
+    for channel in host.api.channels:
+        streams = ["sub", "main", "snapshots"]
+        if host.api.protocol == "rtmp":
+            streams.append("ext")
+
+        for stream in streams:
+            cameras.append(ReolinkCamera(hass, config_entry, channel, stream))
+
+    async_add_devices(cameras, update_before_add=True)
+
+
+class ReolinkCamera(ReolinkCoordinatorEntity, Camera):
+    """An implementation of a Reolink IP camera."""
+
+    _attr_supported_features: CameraEntityFeature = CameraEntityFeature.STREAM
+
+    def __init__(self, hass, config, channel, stream):
+        """Initialize Reolink camera stream."""
+        ReolinkCoordinatorEntity.__init__(self, hass, config)
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Pass the `ReolinkData` instance instead.